code/modules/vending/_vending.dm

```diff
@@ -759,33 +796,51 @@ GLOBAL_LIST_EMPTY(vending_products)
   * * I - the item being loaded
   * * user - the user doing the loading
   */
-/obj/machinery/vending/proc/canLoadItem(obj/item/I,mob/user)
-	return FALSE
-/**
-  * Is the passed in user allowed to load this vending machines compartments
-  *
-  * Arguments:
-  * * user - mob that is doing the loading of the vending machine
-  */
-/obj/machinery/vending/proc/compartmentLoadAccessCheck(mob/user)
-	if(!canload_access_list)
+/obj/machinery/vending/proc/canLoadItem(obj/item/I, mob/user)
+	. = FALSE
+	if(loaded_items >= max_loaded_items)
+		say("There are too many items in stock.")
+		return
+	if(I.custom_price)
 		return TRUE
-	else
-		var/do_you_have_access = FALSE
-		var/req_access_txt_holder = req_access_txt
-		for(var/i in canload_access_list)
-			req_access_txt = i
-			if(!allowed(user) && !(obj_flags & EMAGGED) && scan_id)
-				continue
-			else
-				do_you_have_access = TRUE
-				break //you passed don't bother looping anymore
-		req_access_txt = req_access_txt_holder // revert to normal (before the proc ran)
-		if(do_you_have_access)
-			return TRUE
-		else
-			to_chat(user, "<span class='warning'>[src]'s input compartment blinks red: Access denied.</span>")
-			return FALSE
 
 /obj/machinery/vending/onTransitZ()
 	return
+
+/obj/machinery/vending/custom
+	name = "Custom Vendor"
+	icon_state = "robotics"
+	icon_deny = "robotics-deny"
+	payment_department = NO_FREEBIES
+	refill_canister = /obj/item/vending_refill/custom
+	custom = TRUE
+
+/obj/machinery/vending/custom/Destroy()
+	for(var/obj/item/I in contents)
+		I.forceMove(get_turf(src))
```

Comment:
forceMove doesn't accept nulls. It's also wasteful to do get_turf() for ever single item.
Yes, I know this is copypasta, doesn't mean you don't have to fix it.